pkgs/tools/typesetting/tex/texlive/default.nix

```
@@ -405,6 +407,16 @@ let
             cp texdoc/cache-tlpdb.lua "$out"/scripts/texdoc/Data.tlpdb.lua
           fi
         '';
+
+        # install zsh completion
+        postFixup = ''
+          TEXMFCNF="${bin.core}"/share/texmf-dist/web2c TEXMF="$scriptsFolder/../.." \
+            texlua "$out"/bin/texdoc --print-completion zsh > "$TMPDIR"/_texdoc
+          substituteInPlace "$TMPDIR"/_texdoc \
+            --replace '$(kpsewhich -var-value TEXMFROOT)/tlpkg/texlive.tlpdb' '$(kpsewhich Data.tlpdb.lua)' \
+            --replace '/^name[^.]*$/ {print $2}' '/^  \["[^"]*"\] = {$/ { print substr($1,3,length($1)-4) }'
```

I can't get this to work.

It looks like this is more intended to be a snippet inside a shellrc instead of a standalone completion file. I [think](https://man.archlinux.org/man/zshcompsys.1.en#Autoloaded_files) the completion file should start with a magic comment `#compdef texdoc` and contain a script that is run whenever texdoc is completed -- So I tried executing the function `__texdoc` defined by the script:
```suggestion
          substituteInPlace "$TMPDIR"/_texdoc \
            --replace 'compdef __texdoc texdoc' '#compdef texdoc' \
            --replace '$(kpsewhich -var-value TEXMFROOT)/tlpkg/texlive.tlpdb' '$(kpsewhich Data.tlpdb.lua)' \
            --replace '/^name[^.]*$/ {print $2}' '/^  \["[^"]*"\] = {$/ { print substr($1,3,length($1)-4) }'
          echo _texdoc >> $TMPDIR/_texdoc
```

But this still doesn't work for me any I'm not really whether I'm just holding it terribly wrong.

---

Thanks for checking, I was afraid of something like this (I only tried `. _texdoc` quickly), let me check.

---

Rebased, it should work now. Just copy `_texdoc` in one of your `$fpath`, run `compinit -u` if not already in your `.zshrc`, and you should get completions.